### DISCUSSION TITLE
Is it a reasonnable and possible usage of slackdump as a library

### DISCUSSION THREAD

**Discussion by ySp-chld** [rusq/slackdump#323] [General]
Hello,
I often find slack, very much lacking in term of notification capabilities. No filter, no regex,  no saved search, I'm looking at a way to download message from some specific channel every 5 minutes and filter based upon my choices to generate or not desktop notification that are relevant and useful to me. 
Would this project be a good way to implement such workflow ? I definitely don't want to use a slackbot or apps it would make no sense to do that if a simple client side can do that. 
Best regards , and nice project

**Comment by rusq** [comment]
Hey, I don't think there's an issue with using it as a library, depending on your needs of course.  If you're looking into scraping some channel on a periodic basis, then I think it would work for you.  To authenticate you could use the exported auth or github.com/rusq/slackauth - and then initialise the new Slackdump instance with the returned "Provider"

**Reply by ySp-chld** [reply]
OK, so I could have a local app running, downloading the latest 5 minutes of messages in some channels every minutes or so and then apply some proper filtering on the data. 
This is super nice to know. Thank you for the info.
